.github/workflows/pr-tests.yml

```diff
@@ -49,8 +56,19 @@ jobs:
       run: git clone https://github.com/0xPolygonHermez/pilcom.git  && cd pilcom && npm install
     - name: Build
       run: cargo build --all-targets --all --all-features --profile pr-tests
+    - name: ⚡ Save rust cache
+      if: github.ref == 'refs/heads/main'
+      uses: actions/cache/save@v3
+      with:
+        path: |
+          ~/.cargo/registry
+          ~/.cargo/git
+          target
+        key: ${{ runner.os }}-cargo-pr-tests-${{ hashFiles('**/Cargo.toml') }}
     - name: Run default tests
+      if: github.ref != 'refs/heads/main'
```

Comment:
I'll merge this to see the effect now, but why did you add this restriction here?

Reply:
Yes, I think it's better to run them again. It's might also be easier to track problems if we ever get unstable tests or so.